src/app/spotlightPage/[spotlightId]/page.tsx

```diff
@@ -0,0 +1,109 @@
+'use client';
+
+import Link from 'next/link';
+import React, { useEffect, useState } from 'react';
+
+import { TourRow, DisplayRow } from '../../../types/types';
+import { fetchTour } from '../../../supabase/tours/queries';
+import NavBar from '../../../components/userComponents/navBar/navBar';
+import { fetchDisplayfromSpotlight, fetchRelatedSpotlightsfromSpotlightId } from '../../../supabase/tour_displays/queries';
+
+/**
+ * @param -.params
+ * @param -.params.spotlightId
+ * @param -.params.params
+ * @param -.params.params.spotlightId
+ * @param -.params.params
+ * @param -.params.params.spotlightId
+ * @returns a spotlight page given a spotlight Id
+ */
+export default function Page({ params }: { params: { spotlightId: string } }) {
+  const [spotlight, setSpotlight] = useState<TourRow>({
+    created_at: "N/A",
+    description: "N/A",
+    id: "0",
+    name: "N/A",
+    spotlight: true,
+    stop_count: 0,
+  });
+
+  const [displays, setDisplays] = useState<DisplayRow[]>([])
+
+  const [relatedSpolights, setRelatedSpotlight] = useState<TourRow[]>([])
+
+  useEffect(() => {
+    /**
+     * @returns data from tour table and display table
+     */
+    async function fetchData() {
```

Comment:
more descriptive comments on internal fetchData logic.  ex: explain variables we are creating/setting and what data we are retrieving from